package-lock.json

```diff
@@ -9,27 +9,35 @@
       "version": "0.26.1",
       "hasInstallScript": true,
       "dependencies": {
+        "@ethersproject/shims": "5.7.0",
         "@fortawesome/fontawesome-svg-core": "1.2.36",
         "@fortawesome/free-regular-svg-icons": "6.4.0",
         "@fortawesome/free-solid-svg-icons": "6.4.0",
         "@fortawesome/react-native-fontawesome": "0.2.7",
         "@hathor/unleash-client": "0.1.0",
         "@hathor/wallet-lib": "1.0.1",
+        "@json-rpc-tools/utils": "1.7.6",
         "@lavamoat/preinstall-always-fail": "2.0.0",
         "@notifee/react-native": "5.7.0",
         "@react-native-async-storage/async-storage": "1.19.0",
+        "@react-native-community/netinfo": "11.3.1",
         "@react-native-firebase/app": "16.7.0",
         "@react-native-firebase/messaging": "16.7.0",
         "@react-navigation/bottom-tabs": "6.5.8",
         "@react-navigation/native": "6.1.7",
         "@react-navigation/stack": "6.3.17",
         "@sentry/react-native": "5.6.0",
+        "@walletconnect/core": "2.12.2",
+        "@walletconnect/react-native-compat": "2.12.2",
+        "@walletconnect/web3wallet": "1.11.2",
         "assert": "2.0.0",
         "buffer": "4.9.2",
         "console-browserify": "1.2.0",
         "crypto-js": "4.1.1",
         "deprecated-react-native-prop-types": "2.3.0",
-        "events": "1.1.1",
+        "ethers": "5.7.2",
+        "events": "3.3.0",
```

Comment:
I had to update this to the latest version as the old version didn't have `events.off`... 

I tested most of the features and they worked, but there is no changelog [in the repo](https://github.com/browserify/events).

The idea of the lib is to match the node.js implementation, so I'll check each method before merging this PR, but until then, this is open to review